src/game_character.h

```diff
@@ -940,7 +940,7 @@ inline bool Game_Character::IsFacingLocked() const {
 }
 
 inline void Game_Character::SetFacingLocked(bool locked) {
-	data()->lock_facing = locked;
+	data()->lock_facing = locked || IsDirectionFixedAnimationType(GetAnimationType());
```

Comment:
Minor: I find it confusing that SetFacingLocked can silently fail now. From such setters I would expect that they "work"

Reply:
This is how RPG_RT does it. To make the interface more sane we would need a careful refactor to ensure the correct behavior is not disturbed.

I say punt for now.